sbt-scalajs-bundler/src/main/scala/scalajsbundler/Webpack.scala

```diff
@@ -100,32 +108,162 @@ object Webpack {
     * @param log Logger
     * @return The generated bundles
     */
-  def bundle(
-    generatedWebpackConfigFile: File,
-    customWebpackConfigFile: Option[File],
-    webpackResources: Seq[File],
-    entries: Seq[(String, File)],
-    targetDir: File,
-    log: Logger
+  def libraries(
+      generatedWebpackConfigFile: File,
+      customWebpackConfigFile: Option[File],
+      webpackResources: Seq[File],
+      entryPointFiles: Seq[(String, File)],
+      entries: Seq[(String, File)],
+      targetDir: File,
+      log: Logger
   ): Seq[File] = {
 
     val configFile = customWebpackConfigFile
       .map(Webpack.copyCustomWebpackConfigFiles(targetDir, webpackResources))
       .getOrElse(generatedWebpackConfigFile)
 
-    log.info("Bundling the application with its NPM dependencies")
+    log.info("Bundling the applications NPM dependencies")
     Webpack.run("--config", configFile.absolutePath)(targetDir, log)
 
-    val bundles =
-      entries.map { case (key, _) =>
-        // TODO Support custom webpack config file (the output may be overridden by users)
-        targetDir / Webpack.bundleName(key)
-      }
-    bundles
+    entries.map {
+      case (k, v) =>
+        targetDir / Webpack.libraryFileName(k)
+    }
   }
 
+  val loaderScript: String =
+    """
+      |var exports = {};
```

Comment:
Yes, that’s even better :+1: 

I turns out that we don’t want Scala.js to emit a CommonJS module when we want to produce a bundle in that way. We could also just use `ModuleKind.NoModule` and remove these hacks. Instead of using the `library': '__scalajsbundler_deps__'` option we could just use `libraryTarget: 'window'`.

The only problem is that scalajs-bundler is also used by people who don’t want to bundle their app with webpack but just want to resolve their dependencies from NPM, and execute their application with Node.js (see #85). And in such a case we do want to still use `ModuleKind.CommonJSModule`.